recipes-bsp/u-boot/u-boot-harden.inc

```diff
@@ -16,19 +33,39 @@ TDX_UBOOT_HARDENING_PATCHES_UPSTREAM = "\
     file://0005-toradex-integrate-CLI-access-protection-upstream.patch \
     file://0006-toradex-add-implementation-of-bootargs-protection.patch \
     file://0007-toradex-integrate-bootargs-protection-upstream.patch \
+    file://0008-toradex-show-message-if-CLI-access-allowed-common.patch \
+    file://0009-toradex-show-message-if-CLI-access-allowed-upstream.patch \
 "
 
 TDX_UBOOT_HARDENING_PATCHES = "${TDX_UBOOT_HARDENING_PATCHES_UPSTREAM}"
 TDX_UBOOT_HARDENING_PATCHES:apalis-imx8 = "${TDX_UBOOT_HARDENING_PATCHES_DOWNSTREAM}"
 TDX_UBOOT_HARDENING_PATCHES:colibri-imx8x = "${TDX_UBOOT_HARDENING_PATCHES_DOWNSTREAM}"
+TDX_UBOOT_HARDENING_PATCHES:verdin-am62 = "${TDX_UBOOT_HARDENING_PATCHES_DOWNSTREAM_AM62}"
+TDX_UBOOT_HARDENING_PATCHES:verdin-am62-k3r5 = "${TDX_UBOOT_HARDENING_PATCHES_DOWNSTREAM_AM62}"
 
 SRC_URI:append = "\
     file://u-boot-harden.cfg \
     ${TDX_UBOOT_HARDENING_PATCHES} \
 "
 
-do_compile:prepend() {
+do_compile:prepend:imx-generic-bsp() {
     if [ "${TDX_IMX_HAB_ENABLE}" = "0" ] && [ "${TDX_UBOOT_HARDENING_ENABLE}" = "1" ]; then
-        bbfatal 'The combination TDX_IMX_HAB_ENABLE = "0" and TDX_UBOOT_HARDENING_ENABLE = "1" is not allowed: the whitelisting feature (part of the hardening) currently relies on HAB/AHAB.'
+        bbfatal 'The combination TDX_IMX_HAB_ENABLE="0" and TDX_UBOOT_HARDENING_ENABLE="1" is not allowed:' \
+                'the whitelisting feature (part of the hardening) currently relies on HAB/AHAB.'
     fi
 }
+
+k3_compile_sanity_checks() {
+    if [ "${TDX_K3_HSSE_ENABLE}" = "0" ] && [ "${TDX_UBOOT_HARDENING_ENABLE}" = "1" ]; then
+        bbfatal 'The combination TDX_K3_HSSE_ENABLE="0" and TDX_UBOOT_HARDENING_ENABLE="1" is not allowed:' \
+                'the whitelisting feature (part of the hardening) currently relies on K3.'
```

Comment:
Is saying "K3" here in the message the correct terminology? Isn't K3 more referring to the machine architecture. I think HSSE would be the equivalent term to HAB/AHAB, right?

Reply:
I'm not 100% sure about the terminology. Maybe @sergioprado could help here. This [page](https://software-dl.ti.com/jacinto7/esd/processor-sdk-rtos-jacinto7/latest/exports/docs/psdk_rtos/docs/user_guide/developer_notes_security_build.html) refers to K3 as a security architecture and to HS-FS/HS-SE as states of a HS device.

If it helps, let me clarify the dependency we have: basically we need a function provided by module `arch/arm/mach-k3/common.c` in the U-Boot source code which is specific to the K3 architecture. Btw, that was the main reason I used the word "K3" in the fatal message.

Reply:
K3 stands for Keystone 3 and refers to a the family of SoCs from TI. As far as I could tell, there is no name for the secure boot implementation on TI chips (as we have HAB/AHAB for NXP chips). So I think it is fair to use K3 here (I have used K3 in variables and in the secure boot documentation as well).

Reply:
@jsrc27 Thanks for raising the question so we could align our understanding of the topic.
@sergioprado Thanks for the clarification.

I'll go ahead and close this thread then.